### ISSUE TITLE
How do I now generate amplify/backend/api/amplifyDatasource/build/schema.graphql

### ISSUE THREAD

**Issue by tommyvn** [aws-amplify/amplify-cli#6422]
I used to run `amplify codegen models` to generate a full schema into `amplify/backend/api/amplifyDatasource/build/schema.graphql` from the one in `amplify/backend/api/amplifyDatasource/schema.graphql`.
Newer versions of the cli no longer update this file when I run `amplify codegen models`, altho both versions have similar console output.

npm version that does generate this file: `@aws-amplify/cli@4.17.2`
npm version I am using now that doesn't generate this file: `@aws-amplify/cli@4.41.2`

I am currently running a docker container with my code mounted in and the old cli installed to generate the complete schema, but it's painful having to switch between the two versions to generate my complete schema. Has this functionality moved? Is this a bug?

**Comment by nikhname** [comment]
@tommyvn are the models only not generated in the build folder?

**Comment by tommyvn** [comment]
The JavaScript files are generated into src, and they look correct as in they match my amplify schema, so that command is certainly doing _something_ with the schema, but the full graphql schema, the one where the annotations like `@connection` are exploded out into graphql connection types, isn't generated, which is what I really need.

**Comment by AaronZyLee** [comment]
The full schema in the `build` folder is generated by graphql transformers. Modelgen uses this package to conduct the schema validation but does not update the full schema. To generate the built schema, you should use `amplify api gql-compile` instead. This command is also triggered automatically when you run `amplify push`.

**Comment by tommyvn** [comment]
when I run gql-compile I get this:
```
(python-venv-3.8.5)➜  appsync-schema git:(presigned-put-content-size) ✗ amplify api gql-compile
Scanning for plugins...
Plugin scan successful
UndeterminedEnvironmentError: Current environment cannot be determined
Use 'amplify init' in the root of your app directory to initialize your project with Amplify
```

I don't want to run `amplify init`, we're using amplify for client side code and deployment of the server side stuff is not managed by amplify but rather by other cloudformation. All I want is the full schema, like `amplify codegen models` used to do.

I understand that maybe getting that built schema was a side effect of the codegen model but it was a very useful one, so is there any other way to get only the built schema without being forced into the entire amplify ecosystem? For reasons too long to get into this app can't be wired up to an aws account, i only need the generated schema.

**Comment by AaronZyLee** [comment]
@tommyvn In your use case do you use the model files as well? Or just use the full schema generated by graphql transformer? If for the former one, we do consider that to decouple codegen from amplify workflow. If for the latter one, however, the graphql-transformer is highly coupled with CLI's workflow. AFAIK transformers cannot be triggered outside Amplify CLI for now. Fortunately we also have a plan for transformer redesign here #6217, which I think is a good place for your feature-request.

**Comment by tommyvn** [comment]
ideally both but i certainly need the complete schema file separately from an amplify deployment.
I don't ship infra thru amplify, my backend has a large number of components that I don't want managed in one place, and build and runtime mechanics that weren't and still aren't possible with amplify, even after the vNext stuff is stable.
My team takes advantage of the annotations like `@connection` etc to build the schema with consistent connection object shapes for example, but the deployment is handed off to another cloudformation template that looks very little like an amplify one.

For now I will pin to the version of the amplify cli to the one where I was able to export the built schema without being forced to also deploy, and I'll watch the issue you linked to move to that machinery for schema generation without deployment once it's ready.

Thanks all for looking into this for me.

**Comment by kaustavghosh06** [comment]
@tommyvn You might want to use `amplify api gql-compile` to generate the schema.gql in the build directory locally instead of `amplify codegen models`. Please let us know if that worked for you. Closing this issue for now.

**Comment by tommyvn** [comment]
no dice:

```
amplify api gql-compile
UndeterminedEnvironmentError: Current environment cannot be determined
Use 'amplify init' in the root of your app directory to initialize your project with Amplify
```

and following thru with `amplify init` leads me to setting up a server side project, which is not an option for me.

**Comment by kaustavghosh06** [comment]
@tommyvn I'm confused. `amplify init` will just initialize your existing server side project/environment which you might already have?

**Comment by tommyvn** [comment]
I too am confused.
You are suggesting that I run `amplify api gql-compile` but this command demands I run `amplify init`.
You then suggest that I initialize a server side amplify project, but I do not want to do this, as I've already said [I don't have a server side amplify project project](https://github.com/aws-amplify/amplify-cli/issues/6422#issuecomment-763890905), I have an appsync api that I need to generate the full schema for.
Before the server side amplify offering existed I used to do this with `amplify codegen models` but as I described in this issue this command no longer works.
I need to generate the complete schema for an appsync api, I cannot tie generation of the schema to command line access to a server side amplify project.

If you have any suggestions as to how I can compile the amplify annotated schema into a full schema *without* an amplify server side project, or indeed even access to an AWS account, I would like to hear those suggestions, but if your solution requires me to generate an amplify project on AWS it's not an option, I cannot do this, the shape of both my project and team will not allow for this.
